.github/workflows/publish-release.yml

```diff
@@ -1,29 +1,79 @@
 name: Publish Release
 
 on:
-  pull_request:
-    types: [closed]
+  workflow_call:
+    secrets:
+      NPM_TOKEN:
+        required: true
 
 jobs:
   publish-release:
     permissions:
       contents: write
-    if: |
-      github.event.pull_request.merged == true &&
-      startsWith(github.event.pull_request.head.ref, 'release/')
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-        with:
-          # We check out the release pull request's base branch, which will be
-          # used as the base branch for all git operations.
-          ref: ${{ github.event.pull_request.base.ref }}
-      - name: Get Node.js version
-        id: nvm
-        run: echo ::set-output name=NODE_VERSION::$(cat .nvmrc)
-      - uses: actions/setup-node@v2
-        with:
-          node-version: ${{ steps.nvm.outputs.NODE_VERSION }}
-      - uses: MetaMask/action-publish-release@v1
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.sha }}
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version-file: '.nvmrc'
+      - uses: MetaMask/action-publish-release@v2
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Install
+        run: |
+          yarn install
+          yarn build
```

Comment:
Whoops - left a build step in here

Reply:
A bunch of this workflow is centered around building, then caching that build, so it wouldn't apply here.

Check out this repository for a more recent example of how to set this up in a repo with no build step: https://github.com/MetaMask/KeyringController/blob/main/.github/workflows/publish-release.yml

Reply:
Ok thanks! Should the root level `on` be:
```
on:
  push:
    branches: [main]
```
as it is in the KeyringController example: https://github.com/MetaMask/KeyringController/blob/main/.github/workflows/publish-release.yml#L3
or
```
on:
  workflow_call:
    secrets:
      NPM_TOKEN:
        required: true
```    
as it is in the `module-template` repo: https://github.com/MetaMask/metamask-module-template/blob/main/.github/workflows/publish-release.yml#L3

Reply:
Oh I see so the one on the `keyringController` runs for each push to `main` but first checks if its a release branch when deciding whether to continue?

Reply:
The module template way of structuring the workflows is perfectly fine here. Sorry for the confusion, that repo was last updated before that restructuring was done.